custom/logistics/api.py

```diff
@@ -250,30 +261,38 @@ def sms_user_sync(self, ilsgateway_smsuser, username_part=None, password=None,
                 user.save()
         return user
 
-    def add_language_to_user(self, logistics_sms_user, domains=None):
-        if not domains:
-            domains = []
+    def _reassign_number(self, user, phone_number):
+        v = VerifiedNumber.by_phone(phone_number, include_pending=True)
+        if v.domain in self._get_logistics_domains():
+            v.delete()
+            user.save_verified_number(self.domain, phone_number, True)
+
+    def add_language_to_user(self, logistics_sms_user):
         domain_part = "%s.commcarehq.org" % self.domain
         username_part = "%s%d" % (logistics_sms_user.name.strip().replace(' ', '.').lower(),
                                   logistics_sms_user.id)
         username = "%s@%s" % (username_part[:(128 - (len(domain_part) + 1))], domain_part)
         user = CouchUser.get_by_username(username)
-        if user and user.language != logistics_sms_user.language:
+        if not user:
+            return
+
+        if user.language != logistics_sms_user.language:
             user.language = logistics_sms_user.language
             user.save()
 
+        logistics_numbers = {apply_leniency(phone_number) for phone_number in logistics_sms_user.phone_numbers}
+        if set(user.phone_numbers) == logistics_numbers:
+            return
+
         for phone_number in user.phone_numbers:
             user.delete_phone_number(phone_number)
 
         if logistics_sms_user.phone_numbers:
             phone_number = logistics_sms_user.phone_numbers[0]
-            user.set_default_phone_number(phone_number)
             try:
-                user.save_verified_number(self.domain, phone_number, True,
-                                          logistics_sms_user.backend)
+                user.save_verified_number(self.domain, phone_number, True)
+                user.set_default_phone_number(apply_leniency(phone_number))
```

Comment:
should this line go before save_verified_number, so that we at least set the phone number to the user even if it won't be verified?